File: 1/1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  var result = [];
  for (let first = 0; first < nums.length - 1; first++) {
    for (let second = first + 1; second < nums.length; second++) {
      if (nums[first] + nums[second] == target) {
        result[0] = first;
        result[1] = second;
        return result;
      }
    }
  }
};
// @lc code=end

File: 1/1.two-sum.c
#include <stdio.h>
#include <stdlib.h>
/*
 * @lc app=leetcode id=1 lang=c
 *
 * [1] Two Sum
 */

// @lc code=start

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *twoSum(int *nums, int numsSize, int target, int *returnSize)
{
    // brute method
    int *ret = (int *)malloc(sizeof(int) * 2);
    for (ret[0] = 0; ret[0] < numsSize - 1; ret[0]++)
    {
        for (ret[1] = ret[0] + 1; ret[1] < numsSize; ret[1]++)
        {
            if (ret[0] + ret[1] == target)
            {
                *returnSize = 2;
                return ret;
            }
        }
    }
    *returnSize = 0;
    return NULL;
}

// @lc code=end
